pkg/controller/perconaservermongodb/external_service.go

```diff
@@ -126,8 +126,10 @@ func (r *ReconcilePerconaServerMongoDB) removeOutdatedServices(ctx context.Conte
 	service := psmdb.ExternalService(cr, replset, cr.Name+"-"+replset.Name)
 
 	svcNames := make(map[string]struct{}, replset.Size)
-	for i := 0; i < int(replset.Size); i++ {
-		svcNames[service.Name+"-"+strconv.Itoa(i)] = struct{}{}
+	if replset.Expose.Enabled {
```

Comment:
When you disable it you will have initializing status and the cluster will never be ready.
❯ kubectl get psmdb
NAME              ENDPOINT                                         STATUS         AGE
my-cluster-name   my-cluster-name-mongos.psmdb.svc.cluster.local   initializing   8m6s